src/main/java/com/productdock/library/inventory/domain/BookSubscriptions.java

```diff
@@ -0,0 +1,36 @@
+package com.productdock.library.inventory.domain;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Slf4j
+public class BookSubscriptions {
+
+    private String bookId;
+
+    private List<String> subscribers;
+
+    public void subscribeUser(String userId) {
+        if (!subscribers.contains(userId)) {
+            subscribers.add(userId);
+        }
+    }
+
+    public void unsubscribeUser(String userId) {
+        subscribers.remove(userId);
+    }
+
+    public boolean checkSubscription(String userId) {
```

Comment:
meaning of word `check` might not be fully clear. Can we use `isUserSubscribed` or something similar?